app/profile/page.tsx

```diff
@@ -1,30 +1,34 @@
-"use client";
-import "./profile.scss";
-import { useEffect } from "react";
-import { fetchUserDetails } from "@/redux/slices/userSlice/userThunk/fetchUserDetails";
-import { getUserName } from "@/redux/slices/userSlice/userSelectors";
-import { useAppDispatch, useAppSelector } from "@/redux/hooks";
+import { cookies } from "next/headers";
+import ProfilePageClient from "./ProfilePageClient";
+import axios from "axios";
+import clientApi from "@/apiClients/clientApi";
 
-export default function ProfilePage() {
-  const username = useAppSelector(getUserName);
-  const dispatch = useAppDispatch();
+async function getUserDetails() {
+  const cookieStore = cookies();
+  const token = cookieStore.get("token")?.value;
+  try {
+    const response = await clientApi("/users/userDetails", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+      withCredentials: true,

```

Comment:
Zamiast doklejać to wszystko do każdego zapytania możesz z łatwością utworzyć specjalne metody w axiosie które będą wykonywane dla kazdego requestu lub response, włącznie z doklejaniem tokenu.

https://axios-http.com/docs/interceptors

Reply:
Plus, ładniejszym sposobem jest uruchamianie clientApi.get('...') niż przekazywanie obiektu z method: "GET" (który swoją drogą jest domyślny z tego co pamietam wiec nie trzeba go przekazywać wgl.